File: 백준/Silver/17266. 어두운 굴다리/어두운 굴다리.js
const fs = require("fs");
const { parse } = require("path");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = parseInt(input[0]);
const M = parseInt(input[1]);
const ls = input[2].split(" ").map(Number);
const dist = [ls[0], N - ls[ls.length - 1]];

for (let i = 1; i < ls.length; i++) {
  const num = Math.ceil((ls[i] - ls[i - 1]) / 2);
  dist.push(num);
}

console.log(Math.max(...dist));

File: 백준/Silver/17266. 어두운 굴다리/어두운 굴다리.py
import sys
import math
input = sys.stdin.readline
N = int(input())
M = int(input())
ls = list(map(int, input().split()))
dist = [ls[0], N-ls[-1]]
for i in range(1, len(ls)):
    num = math.ceil((ls[i] - ls[i - 1]) / 2)
    dist.append(num)
print(max(dist))